src/System.Linq/src/System/Linq/Single.cs

```diff
@@ -59,28 +59,21 @@ public static TSource Single<TSource>(this IEnumerable<TSource> source, Func<TSo
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.predicate);
             }
 
-            using (IEnumerator<TSource> e = source.GetEnumerator())
+            var result = TryGetSingle(source, predicate, out bool found, out bool foundSingle);
```

Comment:
`var` is only allowed for `new`, hard-cast and as-cast expressions.

```suggestion
            bool result = TryGetSingle(source, predicate, out bool found, out bool foundSingle);
```

Reply:
(The fact that this turned out to not be a boolean really reinforces our guideline, I think :smile:)

Reply:
Sorry, a couple `var`s slipped through :). Fixed.